Patch: 63_update_default_server_language.patch
Author: László 'GCS' Böszörményi <gcs@lsc.hu>
Get the default language from the config file, and do not use the
hardcoded English.
Index: mailman-2.1.12/bin/update
===================================================================
--- mailman-2.1.12.orig/bin/update	2009-03-02 20:50:02.000000000 +0100
+++ mailman-2.1.12/bin/update	2009-03-02 20:50:13.000000000 +0100
@@ -120,7 +120,7 @@
     #
     # First, get rid of any lists/<list> template or lists/<list>/en template
     # that is identical to the global templates/* default.
-    for gtemplate in os.listdir(os.path.join(mm_cfg.TEMPLATE_DIR, 'en')):
+    for gtemplate in os.listdir(os.path.join(mm_cfg.TEMPLATE_DIR, mm_cfg.DEFAULT_SERVER_LANGUAGE)):
         # BAW: get rid of old templates, e.g. admlogin.txt and
         # handle_opts.html
         try:
@@ -154,14 +154,14 @@
                 os.unlink(os.path.join(mlist.fullpath(), gtemplate + '.prev'))
         # Match against the lists/<list>/en/* templates
         try:
-            fp = open(os.path.join(mlist.fullpath(), 'en', gtemplate))
+            fp = open(os.path.join(mlist.fullpath(), mm_cfg.DEFAULT_SERVER_LANGUAGE, gtemplate))
         except IOError, e:
             if e.errno <> errno.ENOENT: raise
         else:
             tcksum = Utils.md5_new(fp.read()).digest()
             fp.close()
             if gcksum == tcksum:
-                os.unlink(os.path.join(mlist.fullpath(), 'en', gtemplate))
+                os.unlink(os.path.join(mlist.fullpath(), mm_cfg.DEFAULT_SERVER_LANGUAGE, gtemplate))
         # Match against the templates/* template
         try:
             fp = open(os.path.join(mm_cfg.TEMPLATE_DIR, gtemplate))
